tests/test_parser.py

```diff
@@ -17,11 +17,8 @@
 from .utils import make_full_tz_list
 
 
+@pytest.mark.usefixtures("parser_fixture")
 class TestDateTimeParser:
-    @classmethod
-    def setup_class(cls):
-        cls.parser = parser.DateTimeParser()
-
     def test_parse_multiformat(self, mocker):
         mocker.patch(
```

Comment:
Just a quick question: the reason you chose to use `@pytest.mark.usefixtures("parser_fixture")` over passing in the fixture as a function argument was to cut down the changes that needed to be made right?

Reply:
That and it looks cleaner IMHO.

Reply:
I am wondering if we could add `mocker` to the pytest mark: `@pytest.mark.usefixtures("parser_fixture", "mocker")`. Is it possible to not set the `cls` inside of the fixtures and remove the `self` calls inside of the functions?

Reply:
> I am wondering if we could add mocker to the pytest mark: @pytest.mark.usefixtures("parser_fixture", "mocker")

not that I'm aware, pytest-mock only works as a function argument.

> Is it possible to avoid setting the clsdirectly inside of the fixtures (it could allow us remove the self calls inside of the functions)?

if we go the function argument way, yes. this will get rid of the self calls.

Reply:
This is a good starting point and I agree it may be a bit cleaner than the function approach.